akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/Running.scala

```diff
@@ -97,6 +109,48 @@ private[akka] object Running {
   import InternalProtocol._
   import Running.RunningState
 
+  import scala.concurrent.duration._
+  import akka.actor.typed.scaladsl.AskPattern._
+
+  private val log = setup.log
+
+  def startReplicationStream(state: RunningState[S]): Unit = {
+    setup.activeActive.map { aa =>
+      val query = PersistenceQuery(setup.context.system)
+      aa.allReplicas.zipWithIndex.foreach {
+        case (replica, _) =>
+          if (replica != aa.replicaId) {
+            val seqNr = state.seenPerReplica(replica)
+            val pid = PersistenceId.replicated(aa.aaContext.id, replica)
+            // FIXME use index to get plugin id to allow diff databases per replica
+            // and config section to define the read journal config
+            // https://github.com/akka/akka/issues/29257
+            val replication = query.readJournalFor[EventsByPersistenceIdQuery](LeveldbReadJournal.Identifier)
+
+            implicit val timeut = Timeout(30.seconds)
+            implicit val scheduler = setup.context.system.scheduler
+
+            val source = RestartSource.withBackoff(2.seconds, 10.seconds, randomFactor = 0.2) { () =>
+              replication
+                .eventsByPersistenceId(pid.id, seqNr + 1, Long.MaxValue)
+                .mapAsync(1) { eventEnvelope: EventEnvelope =>
+                  setup.context.self.ask[ReplicatedEventAck.type] { replyTo =>
+                    log.debug(s"{} received replicated event from {}: {}", aa.replicaId, replica, eventEnvelope)
+                    ReplicatedEventEnvelope(eventEnvelope.event.asInstanceOf[ReplicatedEvent[E]], replyTo)
+                  }
+                }
+                // useful with info about which stream is failing, which isn't included in error logging in RestartSource
+                .log(s"${aa.aaContext.id}-replicationStream-${aa.aaContext.replicaId}")
+                .withAttributes(Attributes.logLevels(onElement = LogLevels.Off))
+            }
+
+            // FIXME, kill switch for stopping
+            source.runWith(Sink.ignore)(SystemMaterializer(setup.context.system).materializer)
```

Comment:
Isn't there a Flow.watch operator to make the stream die if a watched actor dies that we could use here?

Reply:
Ah yes i'd just done the kill switch but watch is nicer, thanks

Reply:
Or as Patrik suggested, the ActorFlow.ask already does the watch for us.